airflow/www/views.py

```diff
@@ -776,11 +776,28 @@ def index(self):
                 current_dags = all_dags
                 num_of_all_dags = all_dags_count
 
-            sort_column = DagModel.__table__.c.get(arg_sorting_key)
-            if sort_column is not None:
+            if arg_sorting_key == "last_dagrun":
+                dag_run_subquery = (
+                    session.query(
+                        DagRun.dag_id,
+                        sqla.func.max(DagRun.execution_date).label("max_execution_date"),
```

Comment:
If there is no latest dag run, those dags should be shown last for either asc or desc sort.

Reply:
I have updated the logic accordingly.